pkg/controller/cluster/csr.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * MinIO-Operator - Manage MinIO clusters in Kubernetes
+ *
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+
+	miniov1beta1 "github.com/minio/minio-operator/pkg/apis/miniocontroller/v1beta1"
+	"github.com/minio/minio-operator/pkg/constants"
+
+	certificates "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	csrType           = "CERTIFICATE REQUEST"
+	rsaPrivateKeyType = "RSA PRIVATE KEY"
+)
+
+// newPrivateKey returns randomly generated RSA private key.
+func newPrivateKey() (*rsa.PrivateKey, error) {
+	return rsa.GenerateKey(rand.Reader, constants.DefaultRSAKeySize)
```

Comment:
We should use ECC (ECDSA) private keys. Reasoning:
 - Some legacy systems only support modern symmetric ciphers (AES-GCM, ChaCha20-Poly1305) with ECDSA but not with RSA.
 - Smaller Key/Signature Size -> Less traffic.
 - Constant time implementation.